src/api/form/PDFForm.ts

```diff
@@ -499,6 +508,68 @@ export default class PDFForm {
     return PDFTextField.of(text, text.ref, this.doc);
   }
 
+  /**
+   * Flatten all form fields.
+   * 
+   * Flattening a form field will take the current appearance and make that part
+   * of the pages content stream. All form fields and annotations associated are removed.
+   * 
+   * For example:
+   * ```js
+   * const form = pdfDoc.getForm();
+   * form.flatten();
+   * ```
+   */
+  flatten() {
+    this.updateFieldAppearances();
+
+    const fields = this.getFields();
+    const pages = this.doc.getPages();
+
+    for (const field of fields) {
+      const widgets = field.acroField.getWidgets();
+
+      for (const widget of widgets) {
+        const pageRef = widget.P();
+        const page = pages.find(x => x.ref === pageRef);
+        let ref = widget.getNormalAppearance();
+
+        if (ref instanceof PDFDict) {
+          if (field instanceof PDFCheckBox) {
+            const value = field.acroField.getValue();
+            ref = ref.get(value) as PDFRef;
+          } else if (field instanceof PDFRadioGroup) {
+            const value = field.acroField.getValue();
+
+            ref = ref.has(value)
+              ? ref.get(value) as PDFRef
+              : ref.get(PDFName.of('Off')) as PDFRef;
+          }
+        }
```

Comment:
I think we can streamline the logic a bit and remove some type assumptions:
```suggestion
        let refOrDict = widget.getNormalAppearance();

        if (
          refOrDict instanceof PDFDict &&
          (field instanceof PDFCheckBox || field instanceof PDFRadioGroup)
        ) {
          const value = field.acroField.getValue();
          const ref = refOrDict.get(value) ?? refOrDict.get(PDFName.of('Off'));
          if (ref instanceof PDFRef) refOrDict = ref;
        }

        if (!(refOrDict instanceof PDFRef)) {
          throw new Error(`Failed to extract appearance ref`);
        }
```